File: user-api/go-sdk/pkg/models/shared/organization.go
package shared

type OrganizationAddress struct {
	City         *string `json:"city,omitempty"`
	Country      *string `json:"country,omitempty"`
	PostalCode   *string `json:"postal_code,omitempty"`
	Street       *string `json:"street,omitempty"`
	StreetNumber *string `json:"street_number,omitempty"`
}

type OrganizationTypeEnum string

const (
	OrganizationTypeEnumVendor  OrganizationTypeEnum = "Vendor"
	OrganizationTypeEnumPartner OrganizationTypeEnum = "Partner"
)

type Organization struct {
	Address          *OrganizationAddress  `json:"address,omitempty"`
	Email            *string               `json:"email,omitempty"`
	ID               *string               `json:"id,omitempty"`
	IsUnlicensedOrg  *bool                 `json:"is_unlicensed_org,omitempty"`
	LogoThumbnailURL *string               `json:"logo_thumbnail_url,omitempty"`
	LogoURL          *string               `json:"logo_url,omitempty"`
	Name             *string               `json:"name,omitempty"`
	Phone            *string               `json:"phone,omitempty"`
	PricingTier      *string               `json:"pricing_tier,omitempty"`
	Signature        *string               `json:"signature,omitempty"`
	Symbol           *string               `json:"symbol,omitempty"`
	Type             *OrganizationTypeEnum `json:"type,omitempty"`
	Website          *string               `json:"website,omitempty"`
}

File: user-api/ts-sdk/src/sdk/models/shared/organization.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class OrganizationAddress extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=city" })
  city?: string;

  @SpeakeasyMetadata({ data: "json, name=country" })
  country?: string;

  @SpeakeasyMetadata({ data: "json, name=postal_code" })
  postalCode?: string;

  @SpeakeasyMetadata({ data: "json, name=street" })
  street?: string;

  @SpeakeasyMetadata({ data: "json, name=street_number" })
  streetNumber?: string;
}

export enum OrganizationTypeEnum {
    Vendor = "Vendor",
    Partner = "Partner"
}


export class Organization extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=address" })
  address?: OrganizationAddress;

  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=is_unlicensed_org" })
  isUnlicensedOrg?: boolean;

  @SpeakeasyMetadata({ data: "json, name=logo_thumbnail_url" })
  logoThumbnailUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=logo_url" })
  logoUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=phone" })
  phone?: string;

  @SpeakeasyMetadata({ data: "json, name=pricing_tier" })
  pricingTier?: string;

  @SpeakeasyMetadata({ data: "json, name=signature" })
  signature?: string;

  @SpeakeasyMetadata({ data: "json, name=symbol" })
  symbol?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: OrganizationTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=website" })
  website?: string;
}
